scanner/src/test/java/edu/ucr/cs/riple/scanner/ConfigurationTest.java

```diff
@@ -0,0 +1,122 @@
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/** Includes tests that check wrong configurations are correctly reported. */
+@RunWith(JUnit4.class)
+public class ConfigurationTest {
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  // Just a dummy factory to run the tests with, tests will not even start with a null factory.
+  protected DisplayFactory<Display> factory = values -> new ClassInfoDisplay("Unknown", "Unknown");
+  protected SerializationTestHelper<Display> tester;
+  protected Path root;
+
+  @Before
+  public void setup() {
+    root = Paths.get(temporaryFolder.getRoot().getAbsolutePath());
+  }
+
+  @Test
+  public void checkExceptionIsThrownIfConfigPathNotSet() {
+    // -XepOpt:Scanner:ConfigPath is not set should expect an error.
+    tester =
+        new SerializationTestHelper<>(root)
+            .setArgs(
+                Arrays.asList(
+                    "-d", temporaryFolder.getRoot().getAbsolutePath(), "-Xep:Scanner:ERROR"))
+            .setOutputFileNameAndHeader("Unknown", "Unknown")
+            .addSourceFile("SampleClassForTest.java")
+            .setFactory(factory);
+    tester.doTest(
+        "Error in Scanner Checker configuration, should be set with via error prone flag: (-XepOpt:Scanner:ConfigPath)");
+  }
+
+  @Test
+  public void checkOutputDirIsNonnull() {
+    // <scanner>...<output> tag is not set and should expect an error.
+    Path config = root.resolve("scanner.xml");
+    try {
+      Files.createDirectories(root);
+      Files.createFile(config);
+      DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+      try {
+        // Make empty <scanner> tag with no <output> as child.
+        DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+        Document doc = docBuilder.newDocument();
+        Element rootElement = doc.createElement("scanner");
+        doc.appendChild(rootElement);
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        Transformer transformer = transformerFactory.newTransformer();
+        DOMSource source = new DOMSource(doc);
+        StreamResult result = new StreamResult(config.toFile());
+        transformer.transform(source, result);
+      } catch (ParserConfigurationException | TransformerException e) {
+        throw new RuntimeException("Error happened in writing config.", e);
```

Comment:
in -> while

Reply:
42b84b5fc2f8224b9eb1f8fa29009aa3b579febd